include/mxx/reduction.hpp

```diff
@@ -731,7 +731,7 @@ inline std::vector<T> local_scan(const std::vector<T>& in) {
 // global scans
 
 template <typename InIterator, typename OutIterator, typename Func>
-void global_scan(InIterator begin, InIterator end, OutIterator out, Func func, const mxx::comm& comm = mxx::comm()) {
+void global_scan(InIterator begin, InIterator end, OutIterator out, Func func, const mxx::comm& comm = mxx::comm(), const bool commutative = true) {
```

Comment:
Can you flip the parameters? I use the convention `const mxx::comm& comm = mxx::comm()` as the last parameter for all functions.

Other than that LGTM

Reply:
Will make that change.

This might break existing code, though. Should it be recorded somewhere?